src/bin/languageserver/mod.rs

```diff
@@ -1230,3 +1266,157 @@ fn loc_to_range(loc: &pt::Loc, file: &ast::File) -> Range {
 
     Range::new(start, end)
 }
+
+fn update_file_contents(
+    mut prev_content: String,
```

Comment:
How about `prev_content: &str` here

Reply:
`prev_content` needs to be of type `String` due to the restrictions placed by the `fold` method. `fold` requires the input and output of the function passed to it, to have both input and output of the same type (`String` here). https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fold